### PR TITLE
GDScript: Document static variables

### LINKED ISSUE #7487: State that static variables are supported
**Your Godot version:** 4

**Issue description:** The GDScript reference says that static members are not allowed, but in 4.1 that changed.

> `static`: Defines a static function. Static member variables are not allowed.

**URL to the documentation page (if already existing):** https://docs.godotengine.org/en/latest/tutorials/scripting/gdscript/gdscript_basics.html

### PR DESCRIPTION
**Pull request by dalexeev**
~~Draft because godotengine/godot#77129 hasn't been merged yet.~~

Comments are very welcome.

* Closes #7487.

### CODE REVIEW

File: tutorials/scripting/gdscript/gdscript_basics.rst [hunk 1]
```diff
@@ -1589,6 +1679,19 @@ There are a few things to keep in mind here:
     func _init():
         super(5)
 
+Static constructor
```
Comment (dalexeev) [inline tutorials/scripting/gdscript/gdscript_basics.rst:1682]:
```suggestion
Static initializer
```
?

File: tutorials/scripting/gdscript/gdscript_basics.rst [hunk 2]
```diff
@@ -847,6 +847,96 @@ Valid types are:
...
+        pass
+
+    func _ready():
+        prints(A.x, B.x) # 1 1
+        A.x = 2
+        prints(A.x, B.x) # 2 2
+        B.x = 3
+        prints(A.x, B.x) # 3 3
+
+Static functions can use static member variables. A static variable initializer can call a static function.
```
Comment (NathanLovato) [inline tutorials/scripting/gdscript/gdscript_basics.rst:921]: The first sentence is mentioned before. The second sentence means that you can write something like this, right?

```
static var a = some_static_function(value)
```

If so, it'd be good to have a snippet here.

File: tutorials/scripting/gdscript/gdscript_basics.rst [hunk 3]
```diff
@@ -847,6 +847,96 @@ Valid types are:
...
+
+
+`@static_unload` annotation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Since GDScript classes are resources, having static variables in a script prevents it from being unloaded
+even there are no more instances of that class and no other references left. This can be important
+if static variables store large amounts of data or hold references to other project resources, such as scenes.
+You should clean up this data manually, or use the :ref:`@static_unload <class_@GDScript_annotation_@static_unload>`
+annotation if static variables don't store important data and can be reset.
```
Comment (NathanLovato) [inline tutorials/scripting/gdscript/gdscript_basics.rst:932]: Excellent!

File: tutorials/scripting/gdscript/gdscript_basics.rst [hunk 4]
```diff
@@ -847,6 +847,96 @@ Valid types are:
...
+
+.. warning::
+
+    Currently, due to a bug, scripts are never freed.
```
Comment (NathanLovato) [inline tutorials/scripting/gdscript/gdscript_basics.rst:936]: It'd be good to have a snippet here too after the warning, to document usage.

**Review by NathanLovato — COMMENTED** [review]
Great contribution overall, I left some suggestions that I think would help make this more accessible.

File: tutorials/scripting/gdscript/gdscript_basics.rst (hunk 2 above)
Comment (dalexeev) [reply tutorials/scripting/gdscript/gdscript_basics.rst]: I've removed this as it's more related to #7507.

**Comment by MisoMosiSpy** [comment]
This change also fixes https://github.com/godotengine/godot-docs/issues/7487.

File: tutorials/scripting/gdscript/gdscript_basics.rst [hunk 5]
```diff
@@ -847,6 +847,108 @@ Valid types are:
...
+        A.x = 2
+        prints(A.x, B.x) # 2 2
+        B.x = 3
+        prints(A.x, B.x) # 3 3
+
+``@static_unload`` annotation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Since GDScript classes are resources, having static variables in a script prevents it from being unloaded
+even there are no more instances of that class and no other references left. This can be important
```
Comment (MisoMosiSpy) [inline tutorials/scripting/gdscript/gdscript_basics.rst:934]: I believe this should be.
even **if** there are .....

File: tutorials/scripting/gdscript/gdscript_basics.rst [hunk 6]
```diff
@@ -847,6 +847,108 @@ Valid types are:
...
     the project settings. See :ref:`doc_gdscript_warning_system` for details.
 
+Static variables
+^^^^^^^^^^^^^^^^
+
+A class member variable can be declared static::
+
+    static var a
+
+The ``@export`` and ``@onready`` annotations cannot be applied to a static variable.
```
Comment (MisoMosiSpy) [inline tutorials/scripting/gdscript/gdscript_basics.rst:857]: Does it make sense to move these two lines after the static variable explanation (currently line number 866)? I believe that is better flow of information. Also we can make it as warning or tip?

**Review by MisoMosiSpy — COMMENTED** [review]
Look good to me now, thanks!

**Review by skyace65 — APPROVED** [review]

**Comment by skyace65** [comment]
@NathanLovato any other changes you think need to be made?

**Review by NathanLovato — APPROVED** [review]

**Comment by NathanLovato** [comment]
@skyace65 all good!

**Comment by skyace65** [comment]
Thanks!

**Pull request merged**